bin/repos-lint

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+import yaml
+import socket
+from urllib.parse import urlparse
+from urllib.request import urlopen
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("filename")
+args = parser.parse_args()
+
+product_pattern = re.compile('^[A-z_0-9]+$')
+
+def validate_product(pkg):
+    print('%16s: ' % 'product name', end='')
+    try:
+        validate_string(pkg)
+        validate_product_name(pkg)
+    except ValueError as e:
+        print(e)
+        return False
+
+    print('OK')
+    return True
+
+def validate_string(s):
+    if not type(s) is str:
```

Comment:
I believe the usage of `type()` is correct here because we're imposing limits on what should be contain in the yaml file.  We don't want to allow some sort of serialized object that's a subclass (because that really shouldn't be in the configuration file...).